.gitignore

```diff
@@ -0,0 +1,74 @@
+# Created by https://www.toptal.com/developers/gitignore/api/linux,macos,windows
+# Edit at https://www.toptal.com/developers/gitignore?templates=linux,macos,windows
+
+### Linux ###
+*~
+
+# temporary files which can be created if a process still has a handle open of a deleted file
+.fuse_hidden*
+
+# KDE directory preferences
+.directory
+
+# Linux trash folder which might appear on any partition or disk
+.Trash-*
+
+# .nfs files are created when an open file is removed but is still being accessed
+.nfs*
```

Comment:
These are the sorts of rules that belong in an individual's global gitignore, not a project-specific gitignore like this. Under normal circumstances nfs or fuse files shouldn't be present in this project, and if they are present with a certain user's setup, that individual should add rules to his or her global gitignore.

Reply:
Fuse is actually very common because of [GVfs](https://wiki.gnome.org/Projects/gvfs) (the gnome one, not the Microsoft one) almost all popular Gnome distros (including Ubuntu and Debian derivatives) have it activated. I think it's okay to ignore them at the project level because of their popularity. You are right about NFS though

These lines are here because I just downloaded a common template from gitignore.io without reviewing too much. I'm also not sure about the macOS and Windows ignored files, but I assume is safer to just let them here